### PR TITLE
Add nullable rule

### PR DESCRIPTION
**Pull request by jenssegers**
Currently, the `optional` check allows both `null` and `""`. This validation rule behaves similar to the `optional` one, but only allows `null`. This is especially helpful when validating json input.

### CODE REVIEW

**Comment by mrnnn** [comment]
So sexy <3

File: library/Rules/Nullable.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        if ($input === null) {
+            return true;
+        }
+
+        return parent::check($input);
+    }
+
+    public function validate($input)
+    {
+        if ($this->isOptional($input)) {
```
Comment (WouterSioen) [inline library/Rules/Nullable.php]: I guess this should also be `$input === null` instead of the `isOptional` method call?

**Comment by henriquemoody** [comment]
Thanks for contributing, @jenssegers!

These value is what I recently called `undefined`, I recommend you to check #905 and use the `UndefinedHelper`.

File: library/Exceptions/NullableException.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        ],
+        self::MODE_NEGATIVE => [
+            self::STANDARD => 'The value must not be null',
+            self::NAMED => '{{name}} must not be null',
+        ],
+    ];
+
+    public function chooseTemplate()
+    {
+        return $this->getName() == '' ? static::STANDARD : static::NAMED;
```
Comment (henriquemoody) [inline library/Exceptions/NullableException.php]:
```diff
-- return $this->getName() == '' ? static::STANDARD : static::NAMED;
++ return $this->hasName() ? static::NAMED : static::STANDARD;
```

**Comment by henriquemoody** [comment]
Could you rebase it to `master` and take a look at the [CONTRIBUTING.md](https://github.com/Respect/Validation/blob/master/CONTRIBUTING.md)?

**Comment by jenssegers** [comment]
I applied the suggested changes and added tests.

**Comment by WouterSioen** [comment]
👏

**Comment by henriquemoody** [comment]
I made some changes and I created the PR #909, although I'm not really sure about the name.

How about something like `v::allowNull(v::stringType())->validate($input)`?

**Comment by henriquemoody** [comment]
Closed by #909

Thanks for contributing! 🐼

**Pull request closed without merging**